GooglePayment/src/test/java/com/braintreepayments/api/GooglePaymentUnitTest.java

```diff
@@ -44,40 +53,137 @@
-                    .setCurrencyCode("USD")
-                    .build());
+        mBaseRequest = new GooglePaymentRequest()
+                .transactionInfo(TransactionInfo.newBuilder()
+                        .setTotalPrice("1.00")
+                        .setTotalPriceStatus(WalletConstants.TOTAL_PRICE_STATUS_FINAL)
+                        .setCurrencyCode("USD")
+                        .build());
 
-       GoogleApiAvailability mockGoogleApiAvailability = mock(GoogleApiAvailability.class);
-       when(mockGoogleApiAvailability.isGooglePlayServicesAvailable(any(Context.class))).thenReturn(ConnectionResult.SUCCESS);
+        String configuration = new TestConfigurationBuilder()
+                .googlePayment(new TestGooglePaymentConfigurationBuilder()
+                        .supportedNetworks(new String[]{"AMEX", "VISA"})
+                        .enabled(true))
+                .build();
 
-       mockStatic(GoogleApiAvailability.class);
-       when(GoogleApiAvailability.getInstance()).thenReturn(mockGoogleApiAvailability);
+        mMockFragment = new MockFragmentBuilder()
+                .configuration(configuration)
+                .build();
 
-       ActivityInfo mockActivityInfo = mock(ActivityInfo.class);
-       when(mockActivityInfo.getThemeResource()).thenReturn(2132083045);
+        GoogleApiAvailability mockGoogleApiAvailability = mock(GoogleApiAvailability.class);
+        when(mockGoogleApiAvailability.isGooglePlayServicesAvailable(any(Context.class))).thenReturn(ConnectionResult.SUCCESS);
 
-       mockStatic(ManifestValidator.class);
-       when(ManifestValidator.getActivityInfo(any(Context.class), any(Class.class))).thenReturn(mockActivityInfo);
+        mockStatic(GoogleApiAvailability.class);
+        when(GoogleApiAvailability.getInstance()).thenReturn(mockGoogleApiAvailability);
+
+        ActivityInfo mockActivityInfo = mock(ActivityInfo.class);
+        when(mockActivityInfo.getThemeResource()).thenReturn(2132083045);
+
+        mockStatic(ManifestValidator.class);
+        when(ManifestValidator.getActivityInfo(any(Context.class), any(Class.class))).thenReturn(mockActivityInfo);
+    }
+
+    @Test
+    public void isReadyToPay_sendsIsReadyToPayRequest() throws JSONException {
+        PaymentsClient mockPaymentsClient = mock(PaymentsClient.class);
+        when(mockPaymentsClient.isReadyToPay(any(IsReadyToPayRequest.class))).thenReturn(Tasks.forResult(true));
+
+        mockStatic(Wallet.class);
+        when(Wallet.getPaymentsClient(any(Activity.class), any(Wallet.WalletOptions.class))).thenReturn(mockPaymentsClient);
+
+        GooglePayment.isReadyToPay(mMockFragment, null, new BraintreeResponseListener<Boolean>() {
+            @Override
+            public void onResponse(Boolean aBoolean) {
+                // do nothing
+            }
+        });
+
+        ArgumentCaptor<IsReadyToPayRequest> captor = ArgumentCaptor.forClass(IsReadyToPayRequest.class);
```

Comment:
Question for my own clarification - the `IsReadyToPayRequest` is something we're sending to Google before trying to checkout with Google Pay?

Reply:
We send it to Google in order to check if the customer is set up for Google Pay. I believe it's up to the merchant to decide whether or not to perform this check, but it would certainly be recommended. https://github.com/braintree/braintree-android-google-payment/blob/34109af99bb8ce73eb2fd71cc3e116af304db33f/GooglePayment/src/main/java/com/braintreepayments/api/GooglePayment.java#L127